### PR TITLE
Use a hardcoded template for casedb instances

### PR DESCRIPTION
**Pull request by phillipm**
Cross-requested with 
https://github.com/dimagi/commcare/pull/155
https://github.com/dimagi/javarosa/pull/155

### CODE REVIEW

File: app/src/org/commcare/dalvik/application/CommCareApplication.java [hunk 1]
```diff
@@ -446,6 +449,10 @@ public void initializeAppResources(CommCareApp app) {
         int resourceState;
         try {
             currentApp = app;
+
+            // register data instance logic to handle casedb instances
+            ExternalDataInstanceFactory.registerInstanceBuilder(CaseInstanceTreeElement.MODEL_NAME, new CaseDataInstance());
```
Comment (phillipm) [inline app/src/org/commcare/dalvik/application/CommCareApplication.java]: Sorta awkward place for this, but couldn't think of a better place.

File: app/src/org/commcare/android/util/CommCareUtil.java [hunk 2]
```diff
@@ -87,19 +82,6 @@ public static int countPreds(TreeReference reference) {
...
-            DataModelSerializer s = new DataModelSerializer(bos, new CommCareInstanceInitializer(null));
-            
-            s.serialize(new ExternalDataInstance(instanceRef,"instance"), null);
-            Log.d(TAG, new String(bos.toByteArray()));
-        } catch (IOException e) {
-            // TODO Auto-generated catch block
-            e.printStackTrace();
-        }
-    }
-
```
Comment (phillipm) [inline app/src/org/commcare/android/util/CommCareUtil.java]: Unused

File: app/src/org/commcare/dalvik/application/CommCareApplication.java [hunk 3]
```diff
@@ -446,6 +449,10 @@ public void initializeAppResources(CommCareApp app) {
         int resourceState;
         try {
             currentApp = app;
+
+            // register data instance logic to handle casedb instances
+            ExternalDataInstanceFactory.registerExternalDataInstanceBuilder(CaseInstanceTreeElement.MODEL_NAME, new CaseDataInstance());
```
Comment (phillipm) [inline app/src/org/commcare/dalvik/application/CommCareApplication.java]: Sorta awkward place for this, but couldn't think of a better place.

File: app/src/org/commcare/android/util/CommCareUtil.java (hunk 2 above)
Comment (ctsims) [inline app/src/org/commcare/android/util/CommCareUtil.java]: I often will use this when I am debugging, and would kind of prefer to keep it somewhere accessible if possible.

File: app/src/org/commcare/dalvik/application/CommCareApplication.java (hunk 3 above)
Comment (ctsims) [inline app/src/org/commcare/dalvik/application/CommCareApplication.java]: i'd like to talk about different strategies for putting this in the app

**Comment by phillipm** [comment]
This will need an oldui version too

**Pull request merged**